actions/changelog-generate/action.yaml

```
@@ -21,7 +21,7 @@ inputs:
   config:
     description: Path to the changelog config JSON file
     required: false
-    default: "./.github/changelog-config.json"
+    default: ""
```

What about moving `.github/changelog-config.json` to `actions/changelog-generate/changelog-config.json` and adapting the default like in the suggestion? I think then you could get rid of the "Download default changelog" step completely and also don't need to worry about tags there.

```suggestion
    default: "$GITHUB_ACTION_PATH/changelog-config.json"
```

---

@jakob-ed  @yannick-roeder after testing that i have to say this is definitely a good approach. we should consider this in other use cases. the current use case is different because we use some variable in the config and the config only scrap the data in the repo where it's stored. This causes the URL not be generated. and that is the reason i dont think this is a good match for now

---

So if I understand you correctly, if we would move the `changelog-config.json` file to `actions/changelog-generate/changelog-config.json`, then the template variables, e.g. `${{REPO}}` or `${{OWNER}}` are either empty or not being set correctly then?

---

> we use some variable in the config and the config only scrap the data in the repo where it's stored
Are you sure about this? In 2 out of 3 occurrences of changelog-config.json in this PR (among them actions/changelog-generate/changelog-config.json), there is `https://github.com/bakdata/ci-templates/releases/tag/` hard-coded now. Could that be the reason no variable resolution is happening?

---

correct @yannick-roeder

---

I just noticed we have `${{OWNER}}`, `${{REPO}}` in the changelog-config.json files. It should be `#{{OWNER}}`, `#{{REPO}}` according to the docs: https://github.com/mikepenz/release-changelog-builder-action#template-placeholders 
I guess if you use `${{...}}`, Github CI tries to resolve the templating and so that the `release-changelog-builder-action` can resolve it, it should be `#{{...}}`

---

I was wondering about the discrepancy from the issue and the code as well already

---

Although `${{TO_TAG}}` and `${{TITLE}}` seemed to have worked so far without issues

---

Ah they recently changed this for v4: https://github.com/mikepenz/release-changelog-builder-action/releases/tag/v4.0.0

---

```I was wondering about the discrepancy from the issue and the code as well already```
@yannick-roeder oh yes i agree. i added it here because we have another draft issue and wanted to included it to test it all